tests/rptest/services/http_server.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2020 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import json
+import os
+import signal
+
+from ducktape.cluster.remoteaccount import RemoteCommandError
+from ducktape.services.background_thread import BackgroundThreadService
+
+
+class HttpServer(BackgroundThreadService):
+    """
+    Service wrapping simple HTTP server that logs requests to stdout
+    """
+
+    PERSISTENT_ROOT = "/mnt/http_server"
+    SCRIPT = os.path.join(PERSISTENT_ROOT, "server.py")
+    STDOUT_CAPTURE = os.path.join(PERSISTENT_ROOT, "http_server.stdout")
+
+    logs = {
+        "verifiable_consumer_stdout": {
+            "path": STDOUT_CAPTURE,
+            "collect_default": True
+        },
+    }
+
+    def __init__(self, context, port=8080, stop_timeout_sec=2):
+        super(HttpServer, self).__init__(context, 1)
+        self.port = port
+        self.stop_timeout_sec = stop_timeout_sec
+        self.requests = []
+        # server is running on single node
+        self.url = f"http://{self.nodes[0].account.hostname}:{self.port}"
+
+    def _worker(self, idx, node):
+        node.account.ssh("mkdir -p %s" % HttpServer.PERSISTENT_ROOT,
+                         allow_fail=False)
+
+        server_script = self.render('http_server.py')
```

Comment:
I'd recommend doing something like assuming that our source tree is mounted at `/opt` instead of using templating system. I mean, unless it really is a template. We'll need to come up with a solution for the other tests that do something similar with assuming the source tree is mounted in any case when we move to a clustered deployment.

Reply:
i am happy to move this file into dockerfile, i thought it may be more portable to deliver it via ssh, especially when we move to test in clustered deployment

Reply:
i've end up adding this to Dockerfile

Reply:
We don't need to change this around now, but for reference SchemaRegistryTest has a very similar piece of code for injecting a script over SSH -- we could pull that out into a helper at some point for tests that want to inject remote code.